### ISSUE TITLE
implement an algorithm to derive a class fraction layer from a classification layer

### ISSUE THREAD

**Issue by janzandr** [EnMAP-Box/enmap-box#265]
_Requested by @lutzhutz._

Given a classification at a finer resolution, derive class fractions at a coarser resolution
![image](https://user-images.githubusercontent.com/45030715/201338107-283cd47c-545b-4514-8e45-da6eb74017fa.png)

**Comment by janzandr** [comment]
Hi @thielfab, I thought we already had such an algorithm!? Or maybe we just talked about it in the past. I'm a bit confused, have you an idea?

**Comment by janzandr** [comment]
@thielfab I only find the version that requires an vector as input:
![image](https://user-images.githubusercontent.com/45030715/201338813-ecb69cd7-78be-4725-9b00-8ea0f1a1b78b.png)

**Comment by thielfab** [comment]
Yes, we talked about this https://bitbucket.org/hu-geomatics/enmap-box/issues/1328/class-fraction-layer-from-categorized

**Comment by janzandr** [comment]
Ah yes, and now that I read it, I think my answer is not correct.
